src/backend/debezium/src/main/java/com/example/DebeziumRunner.java

```diff
@@ -474,6 +494,112 @@ public void startEngine(MyParameters myParameters) throws Exception
         		else
             		logger.warn("olrHost is null - skip setting Openlog Replicator property");
 
+				if (myParameters.ispnCacheType != null &&
+					myParameters.ispnMemoryType != null &&
+					myParameters.ispnMemorySize > 0)
+				{
+					String memtype = "HEAP";
+
+					if (myParameters.ispnCacheType == "embedded")
+						props.setProperty("log.mining.buffer.type", "infinispan_embedded");
```

Comment:
/* Add Error Handling for Infinispan Setup */
if (myParameters.ispnCacheType != null && myParameters.ispnMemoryType != null && myParameters.ispnMemorySize > 0) {
    try {
        // Existing Infinispan setup code
        props.setProperty("log.mining.buffer.type", "infinispan_embedded");
        ...
    } catch (Exception e) {
        logger.error("Failed to initialize Infinispan: " + e.getMessage());
        throw new RuntimeException("Infinispan setup failed", e);
    }
}